### ISSUE TITLE
Support unpack detection

### ISSUE THREAD

**Issue by GFriedrich** [grafana/logs-drilldown#675]
It would be great if the explorer could detect whether an `unpack` call is needed to show the log data properly.

**Comment by gtk-grafana** [comment]
@GFriedrich can you provide an examples or expected behavior of this proposed feature? I'm not sure I understand the request.

**Comment by GFriedrich** [comment]
Hi @gtk-grafana,
it is my fault - I've used the wrong keyword here. It should read `unpack` not `unwrap`. Sorry for the confusion.
Right now I've a setup which calls `pack` when ingesting the data to Loki. And it would be nice if the UI could automatically detect if the data is "packed" and extend the query by a `| unpack` stage, so all log lines in the UI show correctly.

**Comment by gtk-grafana** [comment]
@trevorwhitney do you know if there's anyway to detect that a query has been packed at ingest?

**Comment by GFriedrich** [comment]
@gtk-grafana not sure if it helps, but Grafana simply checks for the existence of the `_entry` property in the JSON blob:

https://github.com/grafana/grafana/blob/2f012860344d08029e7ad9ce86f7ec79ceee93ec/public/app/plugins/datasource/loki/lineParser.ts#L24

**Comment by trevorwhitney** [comment]
Yeah, I think looking for `_entry` is our best bet, though it raises an interesting question. Maybe adding a key to structured metadata would be useful here?

**Comment by deepcrawltech** [comment]
This would be a great feature. Or at least an ability to update the query manually with | unpack

**Comment by gtk-grafana** [comment]
@trevorwhitney is this something that the new loki endpoints can help us with?

**Comment by trevorwhitney** [comment]
I think we likely need a design doc / discussion. 

* What are the limitations of checking `_entry` in the JSON blob? 
* Does structured metadata, which would need to be checked per line, reasonably solve for these limitations? 
* Is the desire to automatically `unpack` lines in the UI based on a structured metadata property? And how is that different from checking for `_entry`?
* Or is the desire here to be able to hit and endpoint, like `/detected_fields` with the query in question to see if any of the results are `packed` and to add `unpack` to the query?

**Comment by gtk-grafana** [comment]
> Or is the desire here to be able to hit and endpoint, like /detected_fields with the query in question to see if any of the results are packed and to add unpack to the query?

This is what I was thinking, but I didn't think about it very hard 😄 I'll add this to the standup topics and we can decide if we should drum up a bet/design doc for this
